[WIP] v2_key gitignore

jrafanie opened the pull request:
This should fix the dirty git status for an appliance's `certs/v2_key`.  Instead of v2_key being both a development and production key, it is now clearer that a v2_key.dev represents a dev only key.  The v2_key method will look for the dev one only if the production one doesn't exist in non-production modes.

kbrock commented:
can we hold off on this for a day. I have an alternate solution that I have been wanting to implement when I got time. Think it is time to just do it.

Maybe we could compare the 2 and come up with a hybrid

Fryguy commented:
Regardless...the first commit is a straight up bug.  In fact, @jrafanie thinks that a sporadic bug failure he addressed with the appliance console was really this, but he didn't know it at the time.

jrafanie commented:
Updated PR with confirmation that the original flaky test problems I had with running the miq-password_spec.rb before other tests that need to use miq-password are now fixed with this PR.

kbrock commented:
I like most of this except for the `v2_key(name)` part.

I think that should plug in via the legacy key stuff

kbrock commented:
1. I prefer `File.exist?` checks over `Rails.env.production?`
2. We could just have the developer `cp config/v2_key.dev config/v2_key` like they do for `database.pg.yml`
3. With #4091, we could just `load_legacy_key('v2_key.dev')`. It would be a no-op if the file is not present. We already have `load_legacy('v1_key')` which works much the same way.

kbrock commented on gems/pending/util/miq-password.rb:
```diff
@@ -109,13 +109,21 @@ def self.split(encrypted_str)
 
+  def self.key_root
+    @key_root ||= ENV["KEY_ROOT"]
+  end
```
yay!

kbrock commented on gems/pending/util/miq-password.rb:
```diff
@@ -109,13 +109,21 @@ def self.split(encrypted_str)
 
-  def self.v2_key
-    @v2_key ||= ez_load("#{key_root}/v2_key")
+  def self.v2_key(filename = "v2_key")
```
can we not pass in filename here?

jrafanie replied:
It's needed so we can test the various scenarios you see in the specs.

jrafanie commented:
Will need to rebase once #4097 gets merged

jrafanie commented:
@Fryguy @kbrock Do you think #4108 (force developers to copy the v2_key.dev) is better than this technique?  If so, I'll close this.

kbrock commented:
:+1: I like #4108 better. @Fryguy can you close this and merge that?

after that I will rebase my PR ( #4110) and get my BZ done.

Fryguy commented:
Closing in favor of #4108
